sdk/test-utils/perf/src/utils/profiling.ts

```diff
@@ -0,0 +1,45 @@
+import { Session } from "node:inspector";
+import * as fs from "fs-extra";
+
+export async function runWithCpuProfile(
+  functionToProfile: () => Promise<void>,
+  profileFilePath: string | undefined
+) {
+  const session = new Session();
+  session.connect();
+  session.post("Profiler.enable", () => {
+    session.post("Profiler.start", async () => {
+      // Invoke the logic
+      await functionToProfile();
+      // some time later...
+      session.post("Profiler.stop", (err, { profile }) => {
+        // Write profile to disk, upload, etc.
+        if (!err) {
+          profileFilePath =
+            profileFilePath ?? `./profile/${getFormattedDate()}-perfProgram.cpuprofile`;
```

Comment:
the options allow you to specify a default value, so I would use that rather than putting a default inline here

Reply:
By doing `getFormattedDate()`, I won't have the naming conflicts for the artifacts.
What say?

Reply:
You can still use `getFormattedDate` in the default value calculation for the option, can't you?

Reply:
If I put in the default value calculation, it'll be before the test run. 
With the current way, it'll have the exact time of profile generation. 

Hmmm, in case we ever change our profiling model to generate more than one profile(maybe by doing more iterations), it could run into naming conflicts.